### ISSUE TITLE
Capture stdout / stderr to log files

### ISSUE THREAD

**Issue by AlexDBlack** [KonduitAI/konduit-serving#226]
At present, we rely on Java logging (slf4j). This works quite well for Java code, but it doesn't capture logging/printing from:
* Python code
* C/C++ code (native libraries)
In KS, we'll likely have plenty of both.

If possible, we should capture the whole process stdout/stderr to a file, instead of just capturing java logging only.

**Comment by agibsonccc** [comment]
@AlexDBlack in c/c++ this would be nd4j. This might also involve javacpp. @saudet ? In python execution, we might want to have a baseline library for the backend scripts, otherwise we might need a way of doing better logging on the python executioner. This would be a dl4j issues in this case. cc @farizrahman4u

**Comment by saudet** [comment]
I'm not sure if there's a reliable, portable, way to do it all within a single process, but if we're going to have some sort of watchdog process anyway, we can easily capture the output of any child process from there.

**Comment by farizrahman4u** [comment]
In new python executioner, python exceptions from user code is thrown as java exception (with error message from python):

https://github.com/KonduitAI/deeplearning4j/blob/fr_fastpy/datavec/datavec-python/src/test/java/org/datavec/python/TestPythonExecutioner.java#L311

**Comment by AlexDBlack** [comment]
Right, I'm imagining we'll need a system that works for anything, including 3rd party code.
We can't drop arbitrary text.
The watchdog-type approach as @saudet suggests is probably the easiest solution. I'm pretty sure that's how Apache Spark solves this exact problem (executor launches new process, captures all stdout/stderr, which includes but isn't limited to Java logging).
